pool.go

```diff
@@ -17,7 +17,7 @@ func newBufPool(depth, bufLen int) *bufPool {
 func (p *bufPool) Get() []byte {
 	if p == nil {
 		// functional default: no reuse.
-		return make([]byte, p.blen)
+		return nil
```

Comment:
Hm… this was put here as a safeguard, but it never runs, because it was a “just in case”. It definitely is a nil-pointer deref, but changing this to `return nil` won’t actually fix any problems, because this `Get()` needs to return a byte-slice that has an actual length. Returning `nil` would break things worse than a nil-pointer deref: i.e. it breaks things silently.

I think the only answer here is unfortunately to do something like explicitly panic. 🤔

Reply:
Thanks for the explanation!